airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/SSLCertificateUtils.kt

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.jdbc
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.io.BufferedInputStream
+import java.io.ByteArrayInputStream
+import java.io.IOException
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.nio.file.FileSystem
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import java.security.KeyFactory
+import java.security.KeyStore
+import java.security.KeyStoreException
+import java.security.NoSuchAlgorithmException
+import java.security.PrivateKey
+import java.security.SecureRandom
+import java.security.cert.Certificate
+import java.security.cert.CertificateException
+import java.security.cert.CertificateFactory
+import java.security.spec.InvalidKeySpecException
+import java.security.spec.PKCS8EncodedKeySpec
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.SSLContext
+
+private val LOGGER = KotlinLogging.logger {}
+
+/**
+ * General SSL utilities used for certificate and keystore operations related to secured db
+ * connections.
+ */
+object SSLCertificateUtils {
+
+    private const val PKCS_12 = "PKCS12"
+    private const val X509 = "X.509"
+    private val RANDOM: Random = SecureRandom()
+
+    // #17000: postgres driver is hardcoded to only load an entry alias "user"
+    const val KEYSTORE_ENTRY_PREFIX: String = "user"
+    const val KEYSTORE_FILE_NAME: String = KEYSTORE_ENTRY_PREFIX + "keystore_"
+    const val KEYSTORE_FILE_TYPE: String = ".p12"
+
+    @Throws(
+        IOException::class,
+        CertificateException::class,
+        KeyStoreException::class,
+        NoSuchAlgorithmException::class
+    )
+    private fun saveKeyStoreToFile(
+        keyStore: KeyStore,
+        keyStorePassword: String,
+        filesystem: FileSystem?,
+        directory: String?
+    ): URI {
+        val fs = Objects.requireNonNullElse(filesystem, FileSystems.getDefault())
+        val pathToStore = fs!!.getPath(Objects.toString(directory, ""))
+        val pathToFile =
+            pathToStore.resolve(KEYSTORE_FILE_NAME + RANDOM.nextInt() + KEYSTORE_FILE_TYPE)
+        val os = Files.newOutputStream(pathToFile)
+        keyStore.store(os, keyStorePassword.toCharArray())
+        assert(Files.exists(pathToFile) == true)
+        return pathToFile.toUri()
+    }
+
+    @Throws(IOException::class, InterruptedException::class)
+    private fun runProcess(cmd: String, run: Runtime) {
+        LOGGER.debug { "running [$cmd]" }
+        @Suppress("deprecation") val p = run.exec(cmd)
+        if (!p.waitFor(30, TimeUnit.SECONDS)) {
+            p.destroy()
+            throw RuntimeException("Timeout while executing: $cmd")
+        }
+    }
```

Comment:
No. We should never start an `openssl` process. Use bouncycastle instead. ChatGPT is wonderful at generating this kind of code.